src/textual/widgets/_tree.py

```diff
@@ -307,6 +307,7 @@ def set_label(self, label: TextType) -> None:
         self._updates += 1
         text_label = self._tree.process_label(label)
         self._label = text_label
+        self._tree.call_later(self._tree._refresh_node, self)
```

Comment:
Why can't you refresh right here?

Reply:
Calling refresh directly broke much of `test_disabled.py` in a really fun way (try it and see, it's one of the weirdest breaks I've seen so far); some tests taking an age to complete, one just flat out breaking. I suspect some sort of deadly embrace going on; seemed to happen as the `Tree` in the tests was being spun up.

Reply:
Interesting!